features/bsm/service/impl/src/main/java/org/opennms/netmgt/bsm/service/internal/BusinessServiceManagerImpl.java

```diff
@@ -389,12 +434,29 @@ public Status getOperationalStatus(Edge edge) {
                                   .collect(Collectors.toList());
     }
 
+    @Override
+    public List<Application> getAllApplications() {
+        return applicationDao.findAll().stream()
+                .map(s -> {
+                    Hibernate.initialize(s.getMonitoredServices());
+                    return new ApplicationImpl(this, s);
+                })
+                .collect(Collectors.toList());
+    }
+
     @Override
     public IpService getIpServiceById(Integer id) {
         OnmsMonitoredService entity = getMonitoredServiceEntity(id);
         return new IpServiceImpl(this, entity);
     }
 
+    @Override
+    public Application getApplicationById(Integer id) {
+        OnmsApplication onmsApplication = getApplicationEntity(id);
+        Hibernate.initialize(onmsApplication.getMonitoredServices());
```

Comment:
Is this actually working? I had several issues when trying to initialize objects using `Hibernate.initialize`

Reply:
Yes, this works.